src/Tools/Common/Commands/Utils.cs

```diff
@@ -72,6 +72,57 @@ public static bool ValidateArgumentsForChildProcess(int processId, string name,
             return true;
         }
 
+        /// <summary>
+        /// A helper method for validating --process-id, --name options for collect commands and resolving the process ID and name.
+        /// Only one of these options can be specified, so it checks for duplicate options specified and if there is
+        /// such duplication, it prints the appropriate error message.
+        /// </summary>
+        /// <param name="processId">process ID</param>
+        /// <param name="name">name</param>
+        /// <param name="resolvedProcessId">resolvedProcessId</param>
+        /// <param name="resolvedProcessName">resolvedProcessName</param>
+        /// <returns></returns>
+        public static bool ResolveProcess(int processId, string name, out int resolvedProcessId, out string resolvedProcessName)
+        {
+            resolvedProcessId = -1;
+            resolvedProcessName = name;
+            if (processId == 0 && string.IsNullOrEmpty(name))
+            {
+                Console.Error.WriteLine("Must specify either --process-id or --name.");
```

Comment:
Two things:
1. Its legal to run dotnet-trace collect-linux without either of these args specified so this error seems surprising. I assume getting to this point represents a bug in our code rather than a user mistake on the command-line.
2. Testing will be easier if we can intercept these error outputs. We probably want to throw the new ToolException you just added with the string as the message.

Reply:
Right, I was planning on reusing ResolveProcess for example in https://github.com/dotnet/diagnostics/blob/38749d8c420e103c43950417c1b55ed337b2ea5c/src/Tools/dotnet-stack/ReportCommand.cs#L39-L62. (I forgot 😅), so CollectLinux only invokes this under the condition that one is supplied.

I was planning on rebasing either this one or the other PR (hopefully merging this one first), and then having the other change this function to throw the exception.

Reply:
Actually, given that dotnet-stack/ReportCommand is using a textwriter. I think its easier if the other PR that handles throwing exceptions would add the call and catch to write to `stdError`.